### PR TITLE
Ported to ES6 and rewrote some duplicate code

### PR DESCRIPTION
**Pull request by rafaelklaessen**
Ported to ES6 and rewrote some duplicate code. See commits for more details.

### CODE REVIEW

File: bin/initialize.js [hunk 1]
```diff
@@ -1,299 +1,262 @@
 #!/usr/bin/env node
+'use strict';
 
-const program = require('commander');
```
Comment (rafaelklaessen) [inline bin/initialize.js]: This is a compiled file, ES6 files are in src/.

File: bin/initialize.js [hunk 2]
```diff
@@ -1,299 +1,262 @@
...
 						spinner.text = 'application created successfully';
 						spinner.succeed();
-						console.log(`\t$ cd ${projectname}\n \t$ npm install \n \tHappy hacking ♥`);
+						console.log('\t$ cd ' + projectname + '\n \t$ npm install \n \tHappy hacking \u2665');
 					}, 1000);
-				}
-				else {
-					setTimeout(() => {
+				} else {
+					setTimeout(function () {
```
Comment (rafaelklaessen) [inline bin/initialize.js]: This is a compiled file, ES6 files are in src/

File: bin/initialize.js [hunk 3]
```diff
@@ -1,299 +1,262 @@
...
-					else {
-						let opts = [];
-						let propNames = answers.propNames.split(' ');
+					} else {
+						var opts = [];
+						var propNames = answers.propNames.split(' ');
 
-						for(let count=0; count<numberOfPropTypes; count++) {
-							let propTypeChoice = {
+						for (var count = 0; count < numberOfPropTypes; count++) {
```
Comment (rafaelklaessen) [inline bin/initialize.js]: This is a compiled file, ES6 files are in src/.

File: lib/generate.js [hunk 4]
```diff
@@ -1,186 +1,197 @@
...
+		value: function createComponentFile(re, componentName, answersInner) {
+			var _modFile = file.replace(re, componentName);
+			var __modFile = this.createModFile(_modFile, answersInner, componentName);
+			console.log(__modFile);
+			fs.writeFileSync(path.join(process.cwd(), 'src/components', module, componentName + '.react.js'), __modFile, 'utf-8');
+		}
+
+		/**
+   * createComponentFacade
+  	 * @param {string} module - module name to create a component
```
Comment (rafaelklaessen) [inline lib/generate.js]: This is a compiled file, ES6 files are in src/. This is caused by the compiler.

File: lib/generate.js [hunk 5]
```diff
@@ -1,186 +1,197 @@
...
+   * @param {string} componentName - component name
+   * @param {string} asnwers - options provided when creating component
```
Comment (rafaelklaessen) [inline lib/generate.js]: This typo was already here and I didn't see it, my bad.

File: src/lib/init.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+			});
+		} else {
+			// Resolve all promises async-ly
+			promiseShortcut(this, 'eslintrc.json');
+		}
+		
+		/**
+		 * promiseShortcut() function
+		 * Helper method. Created in order to shorten 2 almost the same promise calls.
+		 * @param {init} t Reference to this. Won't work if you just use this, you have to get the reference from a function parameter.
```
Comment (rafaelklaessen) [inline src/lib/init.js]: Init is the current instance of the init class. Normally, this would be `this`.

**Comment by rajikaimal** [comment]
Compiling makes sense, but maintaining compiled files doesn't. So let's go head with the suggestion shall we ? Any other suggestions ?

**Comment by rafaelklaessen** [comment]
Well as said, we are generating reactJS components that use ES6 as well (without any babel config), and we're targetting NodeJS v6  or higher, which supports ES6, so why does compiling make sense?

**Comment by dhruvdutt** [comment]
Please take a look at this table before removing the compiled es5 code.
It seems NodeJS v6 doesn't support ES6 completely.
[http://node.green/](url)

**Comment by rafaelklaessen** [comment]
NodeJS v6 seems to support ES6 for 99%. The only unsupported things (2) aren't being used in the ES6 code. I think it's best to remove the compiled code. Shall we do this?

**Comment by dhruvdutt** [comment]
@rafaelklaessen Sure! But, I think @rajikaimal would make the final call. And we should keep in mind about that 1%. I can help adding gulp task (if needed) or current npm scripts are fine too.

**Comment by rajikaimal** [comment]
What I'm concerned is not about `ES6` compatibility, I want to clarify why we include compiled/transpiled files into our repository. We can do the process of compilation/transpilation while we are testing this tool or adding new version to `npm` registry. Any comments ?

**Comment by rajikaimal** [comment]
I'm assuming that `src` contains all `ES6` code so that it gets transpiled through `babel` to `dist` folder

**Comment by rafaelklaessen** [comment]
No, I'm just compiling everything from ./src to ./ with Babel

**Comment by rafaelklaessen** [comment]
@dhruvdutt Yea, that 1% that's unsupported is just 2 things that I've never heard of, so I'd say it's safe to just use ES6. And once again, ReactJS itself uses ES6 as well.

@rajikaimal The compiled files are included because I didn't want to mess up stuff with the original folsers. In the way it is now, it just works as before. (Btw: I've tested the tool by putting it in my global node_modules folder as npm test didn't test all code.) If we decide to keep the compiling process, we should indeed remove the compiled files and add them when testing or putting it on npm.

**Comment by rajikaimal** [comment]
So if we keep compiled files in our repository it means something like, keeping compiled java file `.class` file in a repository in the case of a java project, right ?

**Comment by rafaelklaessen** [comment]
Kinda, the point is that we don't  _have_ to compile it (but you do have to compile Java)

**Comment by rajikaimal** [comment]
Alright so, if we have `ES6` code now, we do have to compile for sure right ?

**Comment by rafaelklaessen** [comment]
I'd say that we'd just skip the whole compiling thing as nodeJS v6 supports ES6 almost completely.

**Comment by rafaelklaessen** [comment]
You see, it compiles to ES5, but it's just not necessary.

**Comment by rajikaimal** [comment]
Okay, so one question, do you have any compiled files here `es3` / `es5` in this PR

**Comment by rafaelklaessen** [comment]
Yes, in the current PR everything from ./src is compiled to ./. I think that we should put everything that's currently in ./src to ./ and remove the babel config.

**Comment by rajikaimal** [comment]
If we remove `babel` it will remove backward compatibility. So the point is to write code in `ES6` and compile the source into `ES5` for backward compatibility.

**Comment by rafaelklaessen** [comment]
Yes, but we're talking about a scaffolder that generates ES6 for ReactJS.

**Comment by rafaelklaessen** [comment]
So, why wouldn't  a scaffolder that *generates* ES6 be written in ES6 as well?

**Comment by rajikaimal** [comment]
I'm referring to the source code of this tool itself not the code that this tool generates.

**Comment by rafaelklaessen** [comment]
I know, but I mean, using ES5 for a tool that generates ES6 doesn't make sense.

**Comment by rafaelklaessen** [comment]
As the user that uses it has to have node v6 or higher anyways.

**Comment by rajikaimal** [comment]
Like I mentioned earlier we need **backward compatibility**

**Comment by rajikaimal** [comment]
For an example if a developer is using this tool on Node `v4`.

**Comment by rafaelklaessen** [comment]
If a developer is using Node v4 he can't use the generated code either, so the tool's still useless.

**Comment by rafaelklaessen** [comment]
So the tool'd work, but he still wouldn't be able to use it.

**Comment by rajikaimal** [comment]
The code generated by this tool again should be compiled to `ES5` through `babel`.

**Comment by rafaelklaessen** [comment]
But why aren't we adding a babel config for it then?

**Comment by rajikaimal** [comment]
Take a look at [here](https://github.com/99xt/react-scaffolder/tree/master/templates).
We are using `webpack`, and it's loader for `babel`.

**Comment by rafaelklaessen** [comment]
Okay, let's remove the compiled files then.

**Comment by rafaelklaessen** [comment]
How are we gonna let it autocompile on test or when putting on npm registery?

**Comment by rafaelklaessen** [comment]
And how are we going to do that? Merge this PR and then add the autocompile stuff or...?

**Comment by rajikaimal** [comment]
Like I said, lets add the _prepublish_ command on package.json, so that it will compile before publishing a new version to npm registry. 
You can still commit to the same branch and it'll automatically update this same PR.

**Comment by rajikaimal** [comment]
I created a separate branch on this repo called `es6` can you create a PR on that instead. Let's add `es6` stuff there, before we go for a stable release.

**Comment by rafaelklaessen** [comment]
Okey, but just to be sure, you do realise that at the moment, it's compiling to ES5, and that the only thing we need to do is to remove the compiled files (and create prepublish commands etc)?

**Comment by rajikaimal** [comment]
Exactly, so let's go ahead with that plan shall we ?

**Comment by rafaelklaessen** [comment]
Yes, I'll create a pull request.

**Comment by rajikaimal** [comment]
Moved this PR to `es6` branch with PR #19

**Pull request closed without merging**